crate/pkcs11/provider/src/pkcs11_private_key.rs

```diff
@@ -85,40 +85,43 @@ impl PrivateKey for Pkcs11PrivateKey {
             .read()
             .map_err(|e| {
                 error!("Failed to read DER bytes: {:?}", e);
-                MError::Cryptography("Failed to read DER bytes".to_string())
+                MError::Cryptography("Failed to read DER bytes".to_owned())
             })?
             .clone();
         if !der_bytes.is_empty() {
             return Ok(der_bytes);
         }
-        let sk = backend().find_private_key(SearchOptions::Id(self.remote_id.clone()))?;
+        let sk =
+            backend().find_private_key(SearchOptions::Id(self.remote_id.clone().into_bytes()))?;
         let mut der_bytes = self.der_bytes.write().map_err(|e| {
             error!("Failed to write DER bytes: {:?}", e);
-            MError::Cryptography("Failed to write DER bytes".to_string())
+            MError::Cryptography("Failed to write DER bytes".to_owned())
         })?;
         *der_bytes = sk.pkcs8_der_bytes().map_err(|e| {
             error!("Failed to fetch the PKCS8 DER bytes: {:?}", e);
-            MError::Cryptography("Failed to fetch the PKCS8 DER bytes".to_string())
+            MError::Cryptography("Failed to fetch the PKCS8 DER bytes".to_owned())
         })?;
         Ok(der_bytes.clone())
     }
 
     fn rsa_public_exponent(&self) -> MResult<Vec<u8>> {
         let pkcs8_der_bytes = self.der_bytes.read().map_err(|e| {
             error!("Failed to read DER bytes: {:?}", e);
-            MError::Cryptography("Failed to read DER bytes".to_string())
+            MError::Cryptography("Failed to read DER bytes".to_owned())
         })?;
-        Ok(if !pkcs8_der_bytes.is_empty() {
+        let res = if pkcs8_der_bytes.is_empty() {
+            //TODO: not great but very little chance that 1/ it is different and 2/ it has any effect
+            // we do not want to fetch the key bytes just for this
```

Comment:
just pointing this TODO - check if something is needed

Reply:
Replacing this by an error seems safer.

Reply:
right